frontend/src/layout/navigation-3000/sidepanel/panels/activity/sidePanelActivityLogic.tsx

```diff
@@ -105,7 +105,9 @@ export const sidePanelActivityLogic = kea<sidePanelActivityLogicType>([
                         a.created_at.isAfter(b.created_at) ? a : b
                     )
 
-                    if (!latestNotification.unread) {
+                    const hasUnread = values.notifications.some((ic) => ic.unread)
```

Comment:
But then look at the code below, we're always passing the latest `created_at`. How does this work then? How is it possible that a newer notification can be read, whilst an older one isn't?

Reply:
> we're always passing the latest created_at

yep... we store a bookmark per user of the timestamp they've read up to determine if they have something to return from the API

> How is it possible that a newer notification can be read, whilst an older one isn't?

in this case I think Joe's changelog notifications have some special handling, but regardless the check behavior is wrong right now. if there are any unread messages then when you click mark all as read we should set a new bookmark for you

tbh we could always set a bookmark but seems silly to send one if completely unnecessary